apps/documents/forms.py

```diff
@@ -1,39 +1,52 @@
 from django import forms
 
 from apps.documents.models import Collection
-from apps.service_providers.models import LlmProviderTypes
 
 
 class CollectionForm(forms.ModelForm):
     class Meta:
         model = Collection
-        fields = ["name", "is_index", "llm_provider"]
+        fields = ["name", "is_index", "llm_provider", "embedding_provider_model", "is_remote_index"]
         labels = {
             "is_index": "Create file index",
         }
         help_texts = {
             "is_index": "If checked, the files will be indexed and searchable using RAG",
-            "llm_provider": "This is the LLM provider at which the vector store will be created.",
+            "llm_provider": "The provider whose embedding model will be used for indexing",
+            "embedding_provider_model": "The model used to create embeddings",
         }
 
     def __init__(self, request, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.fields["llm_provider"].queryset = request.team.llmprovider_set.filter(type=LlmProviderTypes.openai).all()
+        self.fields["llm_provider"].queryset = request.team.llmprovider_set.all()
+        self.fields["embedding_provider_model"].widget.template_name = "django/forms/widgets/select_dynamic.html"
 
+        # Alpine.js bindings
         self.fields["is_index"].widget.attrs = {"x-model": "isIndex"}
+        self.fields["is_remote_index"].widget.attrs = {"x-model": "isRemoteIndex"}
+        self.fields["llm_provider"].widget.attrs = {
+            "x-model.number.fill": "selectedLlmProviderId",
+        }
+
         if self.instance.id:
             self.fields["is_index"].widget.attrs["disabled"] = True
-            if self.instance.has_pending_index_uploads():
+            if self.instance.has_pending_index_uploads() or not self.instance.is_remote_index:
                 self.fields["llm_provider"].widget.attrs["disabled"] = True
 
-        if self.instance.is_index:
-            self.fields["llm_provider"].widget.is_required = True
+            if self.instance.is_index:
+                self.fields["embedding_provider_model"].widget.attrs["disabled"] = True
+                self.fields["is_remote_index"].widget.attrs["disabled"] = True
```

Comment:
can you explain this logic please

Reply:
I updated this part a bit with one or two comments. Let me know if it's still unclear. https://github.com/dimagi/open-chat-studio/pull/1715/commits/46b1834ce3605e827fb9009912de82d2f7053012